frontend/src/sagas/semesters.js

```diff
@@ -160,10 +160,17 @@ export function* handleSemesterFormSubmit({ values }) {
             (semesterItem) =>
                 semesterItem.currentSemester === true && semesterItem.id !== values.id,
         );
+        const oldDefaultSemester = state.semesters.semesters.find(
+            (semesterItem) =>
+                semesterItem.defaultSemester === true && semesterItem.id !== values.id,
+        );
         if (values.currentSemester && oldCurrentSemester) {
             oldCurrentSemester.currentSemester = false;
-            const { data } = yield call(axiosCall, SEMESTERS_URL, PUT, oldCurrentSemester);
-            yield put(updateSemesterSuccess(data));
+            yield put(updateSemesterSuccess(oldCurrentSemester));
+        }
+        if (values.defaultSemester && oldDefaultSemester) {
+            oldDefaultSemester.defaultSemester = false;
```

Comment:
Data mutation

Reply:
I need it for set old defaultSemester to false and set new default semester to true without doing call to backend

Reply:
fixed